frontend/src/app/dataset/dataset.component.ts

```diff
@@ -129,31 +178,74 @@ export class DatasetComponent {
       if (this.hasLatencies) {
         data.channels['latencies'] = this.sample.latencies[1];
       }
-      console.log('sending to server....', data);
-      this.sharedService.sendFormData(data, 'stats').subscribe((event: any) => {
-        if (typeof event === 'object') {
-          if (event.body !== undefined && event.body.type === 'stats') {
-            for (const i in event.body.avgs) {
-              this.plotRef.addTrace(
-                this.sample.timestamps,
-                event.body.avgs[i],
-                -1,
-                i + ' running avg',
-                true
-              );
-            }
-            // Plot Stdevs here.
-          }
-        }
-      });
+      this.requestStats(data, channel);
     } else {
+      const tab = channel === 'stdev' ? this.tabNumbers[1] : this.tabNumbers[0];
+      const id = toggleStats
+        ? channel + this.currentOptions
+        : String(this.currentOptions);
+
+      this.dashboard.plot.toArray()[tab].toggleTrace(this.ids.get(id));
+
       this.currentShowing
         .get(String(this.currentOptions))
         .set(channel, !this.currentOn(channel));
-      this.plotRef.toggleTrace(this.ids.get(this.currentOptions));
     }
   }
 
+  /**
+   * Handles all needed operations for requesting stats from the backend
+   * including sending a POST, and receiving and plotting the data.
+   * @param data The traces to send to the backend for which
+   *  stats will be computed.
+   * @param channel The data channel for which the stats were requested.
+   */
+  requestStats(data, channel) {
+    console.log('sending to server....', data);
+    this.sharedService.sendFormData(data, 'stats').subscribe((event: any) => {
+      if (
```

Comment:
The spacing of this if block is very odd.

Reply:
The inside of the block? or is the 'if' itself not tabbed over enough? The linter wants me to do it like that

Reply:
The condition part is what's throwing me off. I feel like that should be indented more.

Reply:
I tabbed them over more and added a /*eslint-disable*/

Reply:
That's much more readable.